crowbar_framework/app/models/service_object.rb

```diff
@@ -142,12 +142,18 @@ def random_password(size = 12)
 #
 # Locking Routines
 #
+  def new_lock(name)
+    Crowbar::Lock::LocalBlocking.new(name: name, logger: @logger)
+  end
+
   def acquire_lock(name)
-    FileLock.acquire(name, logger: @logger)
+    new_lock(name).acquire
   end
 
-  def release_lock(f)
-    FileLock.release(f, logger: @logger)
+  def with_lock(name)
+    new_lock(name).with_lock do
+      yield
+    end
   end
```

Comment:
I did wonder about that - actually I spent quite a bit of agonizing over it, and even reduced the amount of duplication (previously there were 3 methods).  But in the end I decided it was more trouble than it was worth.  I'm open to changing my mind if you disagree, but then I'd require details on how to avoid duplication, e.g. what would the shared module/class be called and where would it live?